### PR TITLE
Adding description of escape rule for Groovy scripts in job script

### PR DESCRIPTION
**Pull request by iptizer**
I made the mistake to not escape the variables correctly, see #1413 . This PR adds a description in the seed job documentation.

### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!

Hopefully not required, this is just a small description, for bigger tasks I'd of course use a feature branch.

- [ x ] Ensure that the pull request title represents the desired changelog entry
- [ x ] Please describe what you did
- [ x ] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [ x ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Did you provide a test-case? That demonstrates feature works or fixes the issue.

Only documentation...

### CODE REVIEW

File: docs/seed-jobs.md [hunk 1]
```diff
@@ -28,6 +28,25 @@ jobs:
...
+jenkins:
+  systemMessage: "Seed job with loop."
+jobs:
+  - script: |
+      jobarray = ['job1','job2']
+      for(currentjob in jobarray)
+      multibranchPipelineJob("^${currentjob}") {
+          branchSources {
+              git {
+                  id = "^${currentjob}"
```
Comment (jetersen) [inline docs/seed-jobs.md:43]: This is also valid groovy...
```suggestion
      multibranchPipelineJob("$currentjob") {
          branchSources {
              git {
                  id = "$currentjob"
```

Comment (iptizer) [reply docs/seed-jobs.md]: Updated the PR. What do you think?

**Comment by jetersen** [comment]
link to see current commit author: https://github.com/jenkinsci/configuration-as-code-plugin/commit/c734b9f5bb940eb80e743bd41648266d962ac5d1.patch

@iptizer 
You might want to set up an email alias on your GitHub account 🥇
https://help.github.com/articles/adding-an-email-address-to-your-github-account/

Or look into fixing your git user.email config 😅  
https://help.github.com/articles/setting-your-commit-email-address-in-git
then you would need to amend your commits: https://stackoverflow.com/a/3042512
after changing author

to get proper credit for your commits 🏆

**Comment by iptizer** [comment]
Sorry for the delay. Used the wrong laptop for this PR. Should be corrected by now.

**Comment by jetersen** [comment]
@iptizer I use this trick to ensure my email is always correct: https://github.com/jetersen/dotfiles/blob/5897142a3744e40b7bb20750cf3bfc034971848c/git/gitconfig#L63-L70

**Pull request merged**